File: Java/Medium/Accepted/last-stone-weight-ii_5.java
class Solution {
    public int lastStoneWeightII(int[] stones) {
        
        int sum = 0;
        for (int stone : stones) {
            sum += stone;
        }
        int n = stones.length;
        int m = sum / 2;
        int[][] dp = new int[n + 1][m + 1];
        for (int i = 1; i <= n; i++) {
            for (int j = 1; j <= m; j++) {
                if (stones[i - 1] <= j) {
                    dp[i][j] = Math.max(dp[i - 1][j], dp[i - 1][j - stones[i - 1]] + stones[i - 1]);
                } else {
                    dp[i][j] = dp[i - 1][j];
                }
            }
        }
        return sum - 2 * dp[n][m];
    }
}

File: Rust/Medium/CompileError/last-stone-weight-ii_5.rs
impl Solution {
    pub fn last_stone_weight_ii(stones: Vec<i32>) -> i32 {
        
        let mut dp = vec![false; 1501];
        dp[0] = true;
        let mut sum = 0;
        for &stone in stones.iter() {
            sum += stone;
            for i in (stone..=sum).rev() {
                dp[i] = dp[i] || dp[i - stone];
            }
        }
        for i in (sum / 2..=sum).rev() {
            if dp[i] {
                return sum - 2 * i;
            }
        }
        0
    }
}